src/shared/fstab-util.c

```diff
@@ -95,7 +95,19 @@ int fstab_filter_options(const char *opts, const char *names,
 
         if (!ret_filtered) {
                 for (const char *word = opts;;) {
-                        const char *end = word + strcspn(word, ",");
+                        const char *end = word;
+
+                        /* Look for an *non-escaped* comma separator. Only commas can be escaped, so "\," is
+                         * the only valid escape sequence, so we can do a very simple test here. */
+                        for (;;) {
+                                size_t n = strcspn(end, ",");
+
+                                end += n;
+                                if (n > 0 && end[-1] == '\\')
+                                        end++;
+                                else
+                                        break;
+                        }
```

Comment:
this feels pretty adhoc too. at the very least this should be a function of its own

Reply:
The function is not allowed here, so a helper which allocates cannot be used. OTOH, a helper that just returns the length would replace 3–4 lines, which doesn't seem worth the trouble.